pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/SegmentGeneratorConfig.java

```diff
@@ -441,53 +336,23 @@ public void setRawIndexCreationColumns(List<String> rawIndexCreationColumns) {
     _rawIndexCreationColumns.addAll(rawIndexCreationColumns);
   }
 
-  // NOTE: Should always be extracted from the table config
-  @Deprecated
-  public void setInvertedIndexCreationColumns(List<String> indexCreationColumns) {
-    Preconditions.checkNotNull(indexCreationColumns);
-    _invertedIndexCreationColumns.addAll(indexCreationColumns);
-  }
-
-  /**
-   * Used by org.apache.pinot.core.realtime.converter.RealtimeSegmentConverter
-   * and text search functional tests
-   * @param textIndexCreationColumns list of columns with text index creation enabled
-   */
-  public void setTextIndexCreationColumns(List<String> textIndexCreationColumns) {
-    if (textIndexCreationColumns != null) {
-      _textIndexCreationColumns.addAll(textIndexCreationColumns);
-    }
-  }
-
   @VisibleForTesting
   public void setRangeIndexCreationColumns(List<String> rangeIndexCreationColumns) {
     if (rangeIndexCreationColumns != null) {
-      _rangeIndexCreationColumns.addAll(rangeIndexCreationColumns);
+      setIndexOn(StandardIndexes.range(), new RangeIndexConfig(2), rangeIndexCreationColumns);
```

Comment:
s/2/the_version_constant

Reply:
Problem here: We don't have access to the constant in `pinot-segment-spi` as it is defined in RangeIndexType, which is in `pinot-segment-local`. I can move it to `RangeIndexConfig`, but I don't like the idea of defining the default version in the config.